### PR TITLE
Update mesh guide intro

### PR DESCRIPTION
**Pull request by jasnonaz**
## What are you changing in this pull request and why?
Tweaking / tightening / clarifying the language in the intro to the Mesh guide.

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/guides/best-practices/how-we-mesh/mesh-1-intro.md [hunk 1]
```diff
@@ -6,19 +6,21 @@ hoverSnippet: Learn how to get started with dbt Mesh
...
+
+This guide will walk you through the concepts and implementation details needed to get started. dbt Mesh is not a single product: it is a pattern enabled by a convergence of several features in dbt:
 
 - **[Cross-project references](/docs/collaborate/govern/project-dependencies#how-to-use-ref)** - this is the foundational feature that enables the multi-project deployments. `{{ ref() }}`s now work across dbt Cloud projects on Enterprise plans.
 - **[dbt Explorer](/docs/collaborate/explore-projects)** - dbt Cloud's metadata-powered documentation platform, complete with full, cross-project lineage.
 - **Governance** - dbt's new governance features allow you to manage access to your dbt models both within and across projects.
-  - **[Groups](/docs/collaborate/govern/model-access#groups)** - groups allow you to assign models to subsets within a project.
+  - **[Groups](/docs/collaborate/govern/model-access#groups)** - groups allow you to gather together nodes in your dbt DAG that are logically connected (such as by functional area) and assign an owner to the group.
```
Comment (jasnonaz) [inline website/docs/guides/best-practices/how-we-mesh/mesh-1-intro.md]: I don't __love__ this framing but I think it's better than what's currently there if anyone wants to take a go at it

**Review by dave-connors-3 — APPROVED** [review]
🦞 !

File: website/docs/guides/best-practices/how-we-mesh/mesh-1-intro.md [hunk 2]
```diff
@@ -6,19 +6,21 @@ hoverSnippet: Learn how to get started with dbt Mesh
...
 
 - **[Cross-project references](/docs/collaborate/govern/project-dependencies#how-to-use-ref)** - this is the foundational feature that enables the multi-project deployments. `{{ ref() }}`s now work across dbt Cloud projects on Enterprise plans.
 - **[dbt Explorer](/docs/collaborate/explore-projects)** - dbt Cloud's metadata-powered documentation platform, complete with full, cross-project lineage.
-- **Governance** - dbt's new governance features allow you to manage access to your dbt models both within and across projects.
-  - **[Groups](/docs/collaborate/govern/model-access#groups)** - groups allow you to assign models to subsets within a project.
+- **Governance** - dbt's governance features allow you to manage access to your dbt models both within and across projects.
+  - **[Groups](/docs/collaborate/govern/model-access#groups)** - groups allow you to gather together nodes in your dbt DAG that are logically connected (such as by functional area) and assign an owner to the group.
```
Comment (runleonarun) [inline website/docs/guides/best-practices/how-we-mesh/mesh-1-intro.md]: I suggest we make this sentence active:
```suggestion
  - **[Groups](/docs/collaborate/govern/model-access#groups)** - With groups, you can organize nodes in your dbt DAG that share a logical connection (for example, by functional area) and assign an owner to the entire group.
```

**Review by runleonarun — COMMENTED** [review]
@jasnonaz One tiny suggestion and then I suggest we 🚢 it!

File: website/docs/best-practices/how-we-mesh/mesh-1-intro.md [hunk 3]
```diff
@@ -6,19 +6,21 @@ hoverSnippet: Learn how to get started with dbt Mesh
...
 
 - **[Cross-project references](/docs/collaborate/govern/project-dependencies#how-to-use-ref)** - this is the foundational feature that enables the multi-project deployments. `{{ ref() }}`s now work across dbt Cloud projects on Enterprise plans.
 - **[dbt Explorer](/docs/collaborate/explore-projects)** - dbt Cloud's metadata-powered documentation platform, complete with full, cross-project lineage.
-- **Governance** - dbt's new governance features allow you to manage access to your dbt models both within and across projects.
-  - **[Groups](/docs/collaborate/govern/model-access#groups)** - groups allow you to assign models to subsets within a project.
+- **Governance** - dbt's governance features allow you to manage access to your dbt models both within and across projects.
+  - **[Groups](/docs/collaborate/govern/model-access#groups)** - groups allow you to gather together nodes in your dbt DAG that are logically connected (such as by functional area) and assign an owner to the group.
```
Comment (runleonarun) [inline website/docs/best-practices/how-we-mesh/mesh-1-intro.md]:
```suggestion
  - **[Groups](/docs/collaborate/govern/model-access#groups)** - With groups, you can organize nodes in your dbt DAG that share a logical connection (for example, by functional area) and assign an owner to the entire group.
```

**Review by runleonarun — COMMENTED** [review]
Oops lost this review. Very small suggestion to make sentence active. Otherwise ready to ship!

**Comment by mirnawong1** [comment]
hey folks, looks like this is good to go so am going to merge this. let me know if you feel otherwise.

**Pull request merged**